src/main/java/pro/trevor/tankgame/state/board/unit/tank/GenericTank.java

```diff
@@ -0,0 +1,54 @@
+package pro.trevor.tankgame.state.board.unit.tank;
+
+import org.json.JSONObject;
+import pro.trevor.tankgame.rule.type.IPlayerElement;
+import pro.trevor.tankgame.rule.type.ITickElement;
+import pro.trevor.tankgame.state.board.IMovable;
+import pro.trevor.tankgame.state.board.Position;
+import pro.trevor.tankgame.state.board.unit.GenericUnit;
+
+import java.util.*;
+
+public class GenericTank<E extends Enum<E> & IAttribute> extends GenericUnit<E> implements IMovable, ITickElement, IPlayerElement {
+
+    protected final String player;
```

Comment:
This being protected smells, especially since it's final and `getPlayer()` exists

Reply:
ACK